numpy/_core/numeric.py

```diff
@@ -2361,7 +2360,8 @@ def within_tol(x, y, atol, rtol):
         x = x * ones_like(cond)
```

Comment:
These two lines and the comment above can go, since we actually broadcast `x` and `y` already above.